### PR TITLE
[TGBOT-296] Create http client customer gateway implementation

### PR DESCRIPTION
**Pull request by Sav4ykVladzislau**
https://schneidervs.atlassian.net/browse/TGBOT-296

### CODE REVIEW

File: http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    @Override
+    public Customer saveCustomer(Customer customer) {
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(endpointConfig.getRestApiUrl()))
+                .POST(HttpRequest.BodyPublishers
+                        .ofString(String.format(
+                                "{\"customerId\":\"%s\"\"name\":\"%s\",\"city\":\"%s\",\"phoneNumber\":\"%s\",\"email\":\"%s\"}",
+                                customer.customerId(), customer.name(), customer.city(), customer.phoneNumber(), customer.email())))
```
Comment (max-java) [inline http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java]: this should be posted as body, maybe map from string ? 
good example here https://stackoverflow.com/questions/66388231/json-object-to-body-publisher

File: web-ui/pom.xml [hunk 2]
```diff
@@ -21,6 +21,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.tutrit</groupId>
+            <artifactId>http-client</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (max-java) [inline web-ui/pom.xml]: this doesn't  belong here

File: web-ui/src/main/java/com/tutrit/webclient/WebClientApplication.java [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@ComponentScan(basePackages = {"com.tutrit"})
```
Comment (max-java) [inline web-ui/src/main/java/com/tutrit/webclient/WebClientApplication.java]: redundant, could be removed

File: web-ui/src/main/java/com/tutrit/webclient/controller/CustomerController.java [hunk 4]
```diff
@@ -14,20 +14,17 @@
 
 @Controller
 public class CustomerController {
-    @Autowired(required = false)
+
```
Comment (max-java) [inline web-ui/src/main/java/com/tutrit/webclient/controller/CustomerController.java:17]: changes to web-ui module are out of scope of current issue

File: web-ui/src/main/resources/templates/customer-form.html [hunk 5]
```diff
@@ -1,14 +1,14 @@
 <!DOCTYPE html>
 <html lang="en" xmlns:th="http://www.thymeleaf.org">
```
Comment (max-java) [inline web-ui/src/main/resources/templates/customer-form.html]: changes to web-ui module are out of scope of current issue

**Review by max-java — CHANGES_REQUESTED** [review]
ptal,

File: web-ui/src/main/java/com/tutrit/webclient/WebClientApplication.java (hunk 3 above)
Comment (Sav4ykVladzislau) [reply web-ui/src/main/java/com/tutrit/webclient/WebClientApplication.java]: I did

File: http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java (hunk 1 above)
Comment (Sav4ykVladzislau) [reply http-client/src/main/java/com/tutrit/httpclient/gateway/CustomerGateway/HttpCustomerGateway.java]: I did

File: web-ui/pom.xml (hunk 2 above)
Comment (Sav4ykVladzislau) [reply web-ui/pom.xml]: I did

File: web-ui/src/main/java/com/tutrit/webclient/controller/CustomerController.java (hunk 4 above)
Comment (Sav4ykVladzislau) [reply web-ui/src/main/java/com/tutrit/webclient/controller/CustomerController.java:17]: I agree with you. I reverted this changes

File: web-ui/src/main/resources/templates/customer-form.html (hunk 5 above)
Comment (Sav4ykVladzislau) [reply web-ui/src/main/resources/templates/customer-form.html]: yes, i agree

**Review by yurasava — DISMISSED** [review]

File: http-client/pom.xml [hunk 6]
```diff
@@ -26,7 +26,17 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.tutrit</groupId>
+            <artifactId>web-ui</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
+            <scope>compile</scope>
```
Comment (max-java) [inline http-client/pom.xml:39]: I doubt we need it here - that could lead to circular dependencies

File: web-ui/src/main/java/com/tutrit/webclient/controller/HomePageController.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package com.tutrit.webclient.controller;
```
Comment (max-java) [inline web-ui/src/main/java/com/tutrit/webclient/controller/HomePageController.java]: doesn't seem to be related to #16, does it?

File: web-ui/src/main/resources/templates/starter.html [hunk 8]
```diff
@@ -0,0 +1,354 @@
+<!DOCTYPE html>
```
Comment (max-java) [inline web-ui/src/main/resources/templates/starter.html]: doesn't seem to be related to https://github.com/Tutritp01/TgBot_WebClient/pull/16, does it?

**Review by max-java — COMMENTED** [review]
minor comments.

**Review by Artem-Metlushko — DISMISSED** [review]

**Review by yurasava — DISMISSED** [review]

**Review by schneidervs — DISMISSED** [review]

**Review by yurasava — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Review by AlexanderYarokhovich — APPROVED** [review]

**Review by max-java — APPROVED** [review]
lgfm

**Review by Artem-Metlushko — APPROVED** [review]
